engine/engine.py

```diff
@@ -113,14 +123,14 @@ def fit(self, epoch, is_training=True):
                 if is_training and batch_idx % 100 == 0:
                     if batch_idx % 500 == 0:
                         recon = fwd_output.output_distribution.reparameterise()
-                        recon = recon.reshape((-1,) + input_data.size()[2:]).detach().numpy()
+                        recon = recon.reshape((-1,) + input_data.size()[2:]).detach().cpu().numpy()
```

Comment:
In a future commit, we should make this engine version to be DVAEPP specific, i.e. make this a derived class specific to DVAEPP and steer it through hydra configs.

Reply:
I agree. Need to think about the structure here.